### PR TITLE
SPRINT:  Upgrade many converters of satellite data bufr2ioda.x to align with JEDI Data Conventions

### LINKED ISSUE #869: Upgrade converter for (CONVERTER NAME) to follow the JEDI Data Conventions
A number of testinput yaml files are used with bufr2ioda.x application and need to be updated to use the JEDI Data Conventions. This is the list that will get merged into staging branch feature/sprint-ioda-converters

- [ ]  bufr_airs_v2.yaml
- [ ]  bufr_amsua_v2.yaml
- [ ]  bufr_atms_v2.yaml
- [ ]  bufr_avhrr_v2.yaml
- [ ]  bufr_cris_v2.yaml
- [ ]  bufr_hirs_v2.yaml
- [ ]  bufr_hrs.yaml
- [ ]  bufr_mhs_v2.yaml
- [ ]  bufr_mhs.yaml
- [ ]  bufr_sevcrs_v2.yaml
- [ ]  bufr_ssmis_v2.yaml

### PR DESCRIPTION
**Pull request by gthompsnJCSDA**
## Description

This PR updates 10 converters for satellite data in prepBUFR format using bufr2ioda.x to match the JEDI Data Conventions. Most of the prior versions of yaml files with "_v2" in their name did not have corresponding ctests especially because the global (GDAS) files of satellite data were too large for the repo.  All of the sample files have been greatly shortened.  The following list of YAML files contain the name of each satellite/platform that was tested.

- [x]  bufr_airs.yaml
- [x]  bufr_amsua.yaml
- [x]  bufr_atms.yaml
- [x]  bufr_avhrr.yaml
- [x]  bufr_cris.yaml
- [x]  bufr_hrs.yaml
- [x]  bufr_iasi.yaml
- [x]  bufr_mhs.yaml
- [x]  bufr_sevcrs.yaml
- [x]  bufr_ssmis.yaml

### Issue(s) addressed

Closes #869 

## Acceptance Criteria (Definition of Done)

Reviews and working validation tests.

## Dependencies

Waiting on the following PRs:
- [x] waiting on JCSDA-internal/ioda#655

## Impact

Should be no impacts provided everything is staged in branch ```feature/sprint-ioda-converters```

### CODE REVIEW

**Review by rhoneyager — APPROVED** [review]
Approved, though the branch has a file conflict that should be addressed before merging (gdas.t18z.1bmhs.tm00.nc).

The matching ioda branch has been merged.

**Comment by gthompsnJCSDA** [comment]
> Approved, though the branch has a file conflict that should be addressed before merging (gdas.t18z.1bmhs.tm00.nc).
> 
> The matching ioda branch has been merged.


A variety of minor issues fixed and I think this will now pass CI-testing.

**Comment by rhoneyager** [comment]
@gthompsnJCSDA There is a conflicting testoutput file. Can you resolve?

@rmclaren, @emilyhcliu, @BenjaminRuston Please review this PR.

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -136,13 +136,29 @@ if( iodaconv_bufr_ENABLED )
     testinput/bufr_tables
     testinput/gdas.t18z.1bmhs.tm00.bufr_d
     testinput/gdas.t00z.1bhrs4.tm00.bufr_d
+    testinput/gdas.t00z.airsev.tm00.bufr_d
+    testinput/gdas.t00z.1bamua.tm00.bufr_d
+    testinput/gdas.t00z.avcsam.tm00.bufr_d
+    testinput/gdas.t00z.atms.tm00.bufr_d
```
Comment (rmclaren) [inline test/CMakeLists.txt:142]: Bufr files are very large... We like to keep them around 200 k or so. You can try the subset_bufr.x utility that Rahul wrote a while ago in order to shrink these files does https://github.com/aerorahul/bufr-tools 

usage: >> subset_bufr.x **<input_file>** **<output_file>** **<percent_of_file_to_keep>**
example: >> `subset_bufr.x input.bufr_d output.bufr_d 10`    # Keeps 10 percent of the messages in the file.

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -136,13 +136,29 @@ if( iodaconv_bufr_ENABLED )
...
     testinput/gdas.t06z.adpsfc.tm00.bufr_d
+    testinput/gdas.t00z.crisf4.tm00.bufr_d
+    testinput/gdas.t00z.mtiasi.tm00.bufr_d
+    testinput/gdas.t00z.sevcsr.tm00.bufr_d
+    testinput/gdas.t00z.ssmisu.tm00.bufr_d
```
Comment (rmclaren) [inline test/CMakeLists.txt:147]: Same as last comment.

**Review by rmclaren — CHANGES_REQUESTED** [review]

**Review by emilyhcliu — APPROVED** [review]

File: test/CMakeLists.txt (hunk 1 above)
Comment (rmclaren) [reply test/CMakeLists.txt:142]: If subset_bufr.x gives you problems (its problematic for some files), Emily has a different way of doing it which might work better.

File: test/CMakeLists.txt [hunk 3]
```diff
@@ -847,6 +871,51 @@ if(iodaconv_bufr_ENABLED)
...
+  ecbuild_add_test( TARGET  test_iodaconv_bufr_airs2ioda
+                    TYPE    SCRIPT
+                    COMMAND bash
+                    ARGS    ${CMAKE_BINARY_DIR}/bin/iodaconv_comp.sh
+                            netcdf
+                            "${CMAKE_BINARY_DIR}/bin/bufr2ioda.x testinput/bufr_airs.yaml"
+                            gdas.t00z.airsev.tm00.nc ${IODA_CONV_COMP_TOL_ZERO}
+                    DEPENDS bufr2ioda.x )
+
+  ecbuild_add_test( TARGET  test_iodaconv_bufr_amsua2ioda
```
Comment (rhoneyager) [inline test/CMakeLists.txt:883]: @gthompsnJCSDA This test is failing with "no such file or directory"

**Review by rhoneyager — CHANGES_REQUESTED** [review]
@gthompsnJCSDA Here are the errors.

File: test/CMakeLists.txt [hunk 4]
```diff
@@ -856,6 +925,33 @@ if(iodaconv_bufr_ENABLED)
...
+  ecbuild_add_test( TARGET  test_iodaconv_bufr_seviri2ioda
+                    TYPE    SCRIPT
+                    COMMAND bash
+                    ARGS    ${CMAKE_BINARY_DIR}/bin/iodaconv_comp.sh
+                            netcdf
+                            "${CMAKE_BINARY_DIR}/bin/bufr2ioda.x testinput/bufr_seviri.yaml"
+                            gdas.t00z.sevcsr.tm00.nc ${IODA_CONV_COMP_TOL_ZERO}
+                    DEPENDS bufr2ioda.x )
+
+  ecbuild_add_test( TARGET  test_iodaconv_bufr_ssmis2ioda
```
[the remark this answers is not in the document]
Comment (gthompsnJCSDA) [reply test/CMakeLists.txt:946]: Thank you Ryan - I just made corrections!

**Review by rhoneyager — APPROVED** [review]

**Pull request merged**